pkg/operator/client/deploy.go

```diff
@@ -365,19 +426,59 @@ func findChartNameAndVersion(chartDir string) (string, string, error) {
 	return chartInfo.ChartName, chartInfo.ChartVersion, nil
 }
 
-func (c *Client) uninstallWithHelm(helmDir string, kotsCharts []v1beta1.HelmChart) error {
-	version := "3"
-	chartsDir := filepath.Join(helmDir, "charts")
+func findChartTgz(dir string) (string, error) {
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read dir")
+	}
+
+	for _, file := range files {
+		if filepath.Ext(file.Name()) == ".tgz" {
```

Comment:
can it be `.tar.gz`?

Reply:
looks like helm will support any extension so long as the data is a tarball: https://github.com/helm/helm/blob/main/pkg/chart/loader/archive.go

in this particular case, I think we are safe because we are the ones that [save the upstream chart archive as `.tgz`](https://github.com/replicatedhq/kots/blob/cbo/sc-74872/implement-the-native-helm-without-kustomize/pkg/helmdeploy/write.go#L42).  we can make this more generic if you want and try to read the file as a tarball regardless of extension?

Reply:
> okay, as long as we have tests that cover this :)

what's the case you're concerned about?  i'll see about adding tests

Reply:
thinking of changing this to be more generic since helm does not search based on extension, but whether the file can be read as a tarball.  we should probably do the same

Reply:
updated in https://github.com/replicatedhq/kots/pull/3858/commits/9afb7206f6c87dd0e7b1961eb2e93c26f612d1b4